app/src/main/java/com/sunasterisk/anticovid_19/utils/JsonExtension.kt

```diff
@@ -0,0 +1,20 @@
+package com.sunasterisk.anticovid_19.utils
+
+import com.sunasterisk.anticovid_19.R
+import com.sunasterisk.anticovid_19.data.model.Country
+import com.sunasterisk.anticovid_19.data.model.Global
+import org.json.JSONArray
+import org.json.JSONException
+
+@Throws(JSONException::class)
+inline fun <reified T> String.parseJsonArray() = with(JSONArray(this)) {
+    List(length()) {
+        let { jsonArray ->
+            when (T::class) {
+                Country::class -> Country(jsonArray.getJSONObject(it)) as T
+                Global::class -> Global(jsonArray.getJSONObject(it)) as T
+                else -> throw JSONException(R.string.error_parse_json.toString())
+            }
+        }
+    }
+}
```

Comment:
1. Dòng 12: viết thế này là `this.let` à?
2. `this` nhập nhằng. Lúc `this` là `String`, trong `with` thì `this` là `JSONArray`
3. Dòng 11: `length()` hiểu là this.length() là của `JSONArray.length()`, thì nếu viết thiếu cặp ngoặc `()` thành `length` hiểu là `this.length` thì lại thành `String.length`
=> lạm dụng `scope function`
```suggestion
inline fun <reified T> String.parseJsonArray() = JSONArray(this).let { jsonArray ->
    List(jsonArray.length()) { index ->
        when (T::class) {
            Country::class -> Country(jsonArray.getJSONObject(index)) as T
            Global::class -> Global(jsonArray.getJSONObject(index)) as T
            else -> throw JSONException(R.string.error_parse_json.toString())
        }
    }
}
```

Reply:
Ok tks chú nhiều

Reply:
Đôi lúc dùng nhiều scope function lại khiến đoạn code của mình khó hiểu hơn => Sai mục đích của scope function